helix-view/src/document.rs

```diff
@@ -839,8 +845,10 @@ impl Document {
         let changes = std::mem::replace(&mut self.changes, new_changeset);
         // Instead of doing this messy merge we could always commit, and based on transaction
         // annotations either add a new layer or compose into the previous one.
-        let transaction =
-            Transaction::from(changes).with_selection(self.selection(view_id).clone());
+        let mut transaction = Transaction::from(changes);
+        if let Some(view_id) = view_id {
+            transaction = transaction.with_selection(self.selection(view_id).clone());
+        }
```

Comment:
I think this isn't correct, the code assumes there's always a selection so we have the previous selection stored when undoing. The editor will probably panic otherwise.